### ISSUE TITLE
Migrations error when pushing db

### ISSUE THREAD

**Issue by jeremypeters** [supabase/supabase#17486]
# Bug report

- [ x] I confirm this is a bug with Supabase, not with my own application. 
- [ x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I'm developing locally and trying to push the changes.

When running `supabase db push` I get the error:

`supabase_migrations.schema_migrations table conflicts with the contents of supabase/migrations.; Found 5 versions and 1 migrations.`

Prior steps:

- Due to PKCE error (see [here](https://github.com/supabase/auth-helpers/issues/634)) I created a `gotrue-version` file as per these instructions [here](https://github.com/supabase/auth-helpers/issues/569#issuecomment-1703159422)
- Was then getting "Failed to load configuration: required key API_EXTERNAL_URL missing value" error when trying to use the supabase CLI (see [here](https://github.com/supabase/auth-helpers/issues/569#issuecomment-1720043083))
- Removed the  `gotrue-version` file and restarted supabase
- Ensue the project was linked `supabase link --project-ref xxxxxxxxx`
- Created a migration file using the command `supabase db diff --use-migra -f  misc`

**Comment by jeremypeters** [comment]
I found this [Reddit thread](https://www.reddit.com/r/Supabase/comments/13j8rdl/removing_remote_migrations/) which pointed me in the right direction. It turns out the issue was that my previous migration files had been deleted - not sure how that happened. Anyway I deleted all the entries in the cloud `supabase_migrations.schema_migrations` table.  I then ran `supabase db diff --use-migra -f  misc` and then `supabase db push` again - I no longer got the migrations error, but got errors regarding existing tables. I then created a new cloud Supabase project, linked my local to that and pushed - everything is ok now.

**Comment by jeremypeters** [comment]
I found this [Reddit thread](https://www.reddit.com/r/Supabase/comments/13j8rdl/removing_remote_migrations/) which pointed me in the right direction. It turns out the issue was that my previous migration files had been deleted - not sure how that happened. Anyway I deleted all the entries in the cloud supabase_migrations.schema_migrations table. I then ran supabase db diff --use-migra -f  misc and then supabase db push again - I no longer got the migrations error, but got errors regarding existing tables. I then created a new cloud Supabase project, linked my local to that and pushed - everything is ok now.
